### PR TITLE
Document planned release cadence for arrow/parquet/etc

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Related to https://github.com/apache/arrow-rs/issues/5368

# Rationale for this change
 
We spent a while discussing release cadence in https://github.com/apache/arrow-rs/issues/5368 so let's document that clearly on in the readmes

# What changes are included in this PR?

Document the planed release cadence. 

Add links in parquet / arrow-rs readme

# Are there any user-facing changes?

Documentation, no functional changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,7 +34,33 @@ This repo contains the following main components:
...
+## Release Versioning and Schedule
+
+The Arrow Rust project releases approximately monthly and follows [Semantic
+Versioning](https://semver.org/).
+
+Due to available maintainer and testing bandwidth, `arrow` crates (`arrow`,
+`arrow-flight`, etc.) are released on the same schedule with the same versions
+as the `parquet` and `parquet-derive` crates.
+
+Starting June 2024, we plan to release new major versions with potentially
```
Comment (alamb) [inline README.md:46]: I believe this accurately summarizes the plan from https://github.com/apache/arrow-rs/issues/5368

File: README.md [hunk 2]
```diff
@@ -34,7 +34,33 @@ This repo contains the following main components:
...
+the intervening months. See [this ticket] for more details.
+
+For example:
+
+| Approximate Date | Version  | Notes                                   |
+| ---------------- | -------- | --------------------------------------- |
+| Jun 2024         | `52.0.0` | Major, potentially breaking API changes |
+| Jul 2024         | `52.1.0` | Minor, NO breaking API changes          |
+| Aug 2024         | `52.2.0` | Minor, NO breaking API changes          |
+| Sep 2024         | `52.0.0` | Major, potentially breaking API changes |
```
Comment (GPSnoopy) [inline README.md:57]: Potential typo: likely should be `53.0.0`

Comment (alamb) [reply README.md:57]:
```suggestion
| Sep 2024         | `53.0.0` | Major, potentially breaking API changes |
```

Comment (alamb) [reply README.md]: Thank you -- good eyes -- I fixed that

**Comment by Xuanwo** [comment]
Do we have a plan to object_store? I'm guessing a breaking change to `object_store` could also affect parquet?

**Comment by tustvold** [comment]
> Do we have a plan to object_store? I'm guessing a breaking change to object_store could also affect parquet?

Nothing formal, but a natural implication of this would be that if we made a breaking release to object_store we would be unable to incorporate this into arrow until the next arrow breaking release. In practice this means object_store will likely follow a similar cadence for breaking releases, if not slower.

File: README.md [hunk 3]
```diff
@@ -34,7 +34,33 @@ This repo contains the following main components:
...
+
+The Arrow Rust project releases approximately monthly and follows [Semantic
+Versioning](https://semver.org/).
+
+Due to available maintainer and testing bandwidth, `arrow` crates (`arrow`,
+`arrow-flight`, etc.) are released on the same schedule with the same versions
+as the `parquet` and `parquet-derive` crates.
+
+Starting June 2024, we plan to release new major versions with potentially
+breaking API changes once a quarter, and release incremental minor versions in
```
Comment (tustvold) [inline README.md:47]:
```suggestion
breaking API changes at most once a quarter, and release incremental minor versions in
```

Let's be optimistic :laughing:

**Review by tustvold — APPROVED** [review]

**Comment by alamb** [comment]
I plan to add a note about object_store release schedules, likely as a follow on PR

**Comment by alamb** [comment]
I am working on a follow up PR for object_store cadence

File: README.md [hunk 4]
```diff
@@ -34,7 +34,33 @@ This repo contains the following main components:
 
 The current development version the API documentation in this repo can be found [here](https://arrow.apache.org/rust).
 
-There are two related crates in a different repository
+## Release Versioning and Schedule
+
+The Arrow Rust project releases approximately monthly and follows [Semantic
+Versioning](https://semver.org/).
```
Comment (amoeba) [inline README.md:40]: Hey @alamb, sorry I missed this when it was open.

I am a little confused by this statement and others here. My understanding from https://github.com/apache/arrow-rs/issues/5368 and elsewhere was that arrow-rs doesn't follow semver, except maybe with its minor and patch releases. My understanding of the spirit of semver is that major releases absolutely include breaking changes so if arrow-rs releases majors on a schedule and those majors only _may_ include breaking changes, that wouldn't be following semver. Does that match your interpretation?

Comment (alamb) [reply README.md:40]: > I am a little confused by this statement and others here. My understanding from https://github.com/apache/arrow-rs/issues/5368 and elsewhere was that arrow-rs doesn't follow semver, except maybe with its minor and patch releases. 

From my perspective arrow-rs follows semver as implemented by rust / cargo. 

> My understanding of the spirit of semver is that major releases absolutely include breaking changes 

That may be the "spirit" or ideal, but my understanding is that semver means breaking changes only occur in major versions. 

> so if arrow-rs releases majors on a schedule and those majors only may include breaking changes, that wouldn't be following semver. Does that match your interpretation?

No. 

However, what we have done in the past is if a scheduled release includes no breaking changes, we did a minor release. However, the API is so large practically all releases have had at least one breaking change (I think I remember one that did not) 

what I expect to happen is that each quarterly release will include breaking API changes (however small). If they don't have breaking changes we can do a minor relese

Comment (amoeba) [reply README.md:40]: That makes sense, thanks. I think the fuzziness around scheduled releases having a high likelihood of including breaking changes was where I was confused. Thanks for the quick response!

**Pull request merged**